dev/src/cli/cli_create.ts

```diff
@@ -0,0 +1,218 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as path from 'path';
+import { isCancel, select, text } from '@clack/prompts';
+import { exec, execSync } from 'child_process';
+import { promisify } from 'util';
+import { createFolder, isFolderExists, removeFolder, saveToFile } from '../utils/file_utils';
+
+const execPromise = promisify(exec);
+const dirname = process.cwd();
+
+const AGENT_TEMPLATE = (model: string) => `
+import {FunctionTool, LlmAgent} from '@google/adk';
+import {z} from 'zod';
+
+/* Mock tool implementation */
+const getCurrentTime = new FunctionTool({
+  name: 'get_current_time',
+  description: 'Returns the current time in a specified city.',
+  parameters: z.object({
+    city: z.string().describe("The name of the city for which to retrieve the current time."),
+  }),
+  execute: ({city}) => {
+    return {status: 'success', report: \`The current time in \$\{ city \} is 10: 30 AM\`};
+  },
+});
+
+export const rootAgent = new LlmAgent({
+  name: 'hello_time_agent',
+  model: '${model}',
+  description: 'Tells the current time in a specified city.',
+  instruction: \`You are a helpful assistant that tells the current time in a city.
+                Use the 'getCurrentTime' tool for this purpose.\`,
+  tools: [getCurrentTime],
+});
+`.trim()
+
+interface AgentCreationOptions {
+  agentName: string;
+  model: string;
+  apiKey: string;
+  project: string;
+  region: string;
+  language: string;
+}
+
+async function getGcpProject(): Promise<string> {
+  if (process.env.GOOGLE_CLOUD_PROJECT) {
+    return process.env.GOOGLE_CLOUD_PROJECT;
+  }
+  try {
+    const stdout = execSync('gcloud config get-value project', { encoding: 'utf-8', stdio: 'pipe' });
+    return stdout.trim();
+  } catch (error) {
+    return '';
+  }
+}
+
+async function getGcpRegion(): Promise<string> {
+  if (process.env.GOOGLE_CLOUD_LOCATION) {
+    return process.env.GOOGLE_CLOUD_LOCATION;
+  }
+  try {
+    const stdout = execSync('gcloud config get-value compute/region', { encoding: 'utf-8', stdio: 'pipe' });
+    return stdout.trim();
+  } catch (error) {
+    return '';
+  }
+}
+
+async function generateAgentFolder(agentDir: string) {
+  if (!(await isFolderExists(agentDir))) {
+    await createFolder(agentDir)
+  } else {
```

Comment:
nit to reduce condition nesting:
```suggestion
if (!(await isFolderExists(agentDir))) {
  return createFolder(agentDir)
}

const overwriteFolderResponse = await select({
    message: `Folder ${agentDir} already exists. Would you like to overwrite existing folder?`,
    options: [
      { label: 'Yes', value: true },
      { label: 'No', value: false },
    ],
  },
) as boolean;

if (isCancel(overwriteFolderResponse)) {
  process.exit(0);
}

if (!overwriteFolderResponse) {
  console.error(`Agent directory ${agentDir} already exists.`);
  return;
}

await removeFolder(agentDir);
await createFolder(agentDir);
```